tracing-core/src/field.rs

```diff
@@ -748,7 +748,7 @@ impl<'a> ValueSet<'a> {
     /// Visits all the fields in this `ValueSet` with the provided [visitor].
     ///
     /// [visitor]: super::Visit
-    pub(crate) fn record(&self, visitor: &mut dyn Visit) {
+    pub fn record(&self, visitor: &mut dyn Visit) {
```

Comment:
In general, I would prefer to avoid adding public `tracing-core` APIs. In this case, however, I think `record` should _probably_ be made public anyway, and not making it public was a mistake. So, I'm okay with this.

Reply:
I can keep this, however there is another option that I'd like your opinion on. I could change the valueset macro to output a slice instead of a valueset, that is remove this:
```rust
        {
            #[allow(unused_imports)]
            use $crate::field::{debug, display, Value};
            let mut iter = $fields.iter();
            $fields.value_set($crate::valueset!(
                @ { },
                iter.next().expect("FieldSet corrupted (this is a bug)"),
                $($kvs)+
            ))
        }
```
Then I could generate the log from the slice instead of the valueset, which would dramatically simplify the code.

Reply:
I've left the implementation the way it is, it turned out my suggestion didn't really help because it still had to go through `Value::record`